src/index.tsx

```diff
@@ -1,22 +1,41 @@
 import * as React from "react";
 import { render } from "react-dom";
 
-import { ApolloClient, ApolloProvider, useQuery, InMemoryCache } from "@apollo/client";
-import { GET_CHARACTER } from "./queries";
-import { GetCharactersQuery } from "./common/graphql/types/graphql";
+import {
+  ApolloClient,
+  ApolloProvider,
+  useQuery,
+  InMemoryCache,
+} from "@apollo/client";
+import { GET_CHARACTER, CharactersCountFragment } from "./queries";
+import { FragmentType, useFragment } from "./common/graphql/types";
 
 const client = new ApolloClient({
   uri: "https://rickandmortyapi.com/graphql",
   cache: new InMemoryCache(),
 });
 
+type CharactersCountProps = {
+  characters: FragmentType<typeof CharactersCountFragment>;
```

Comment:
It means that we can also generate prop types too.

We know that `characters` will be `FragmentType<typeof CharactersCountFragment>` and that unneeded properties of `characters` will be masked. **We can't access properties we don't explicitly query for, even though they are available!**